onnxruntime/core/providers/cuda/tensor/split.cc

```diff
@@ -76,6 +76,28 @@ Status SplitKernel::ComputeInternal(OpKernelContext* ctx) const {
   auto input_dims = input_shape.GetDims();
   auto output_dimensions{input_shape.AsShapeVector()};
 
+  if (split_sizes.size() == 3 && (block_size_inside_axis_dim == 1)) {
+    // we use block_size_inside_axis_dim == 1 to check if we are splitting on inner most axis.
+    // only when split on inner axis and output size is 3, we can use Split3Inner.
```

Comment:
Would it be better to introduce a new attribute? @tianleiwu what do you think?

Reply:
adding a new attribute will need to set it when using this feature. but most times, we just export the model from torch, and don't have the opportunity to set this attribute. So I think we can leave it like this.

Reply:
@wangyems, Split is in onnx domain and it is not easy to add attribute.